app/models/spree/user.rb

```diff
@@ -144,6 +144,32 @@ def flipper_id
       "#{self.class.name};#{id}"
     end
 
+    def disabled?
+      disabled_at.present?
+    end
+
+    def disable!
+      self.disabled_at = Time.zone.now
+      save!
+    end
+
+    def enable!
+      self.disabled_at = nil
+      save!
+    end
+
+    def toggle_disable
+      disabled?
+    end
+
+    def toggle_disable=(value)
+      if value == '1'
+        disable!
+      else
+        enable!
+      end
+    end
+
```

Comment:
I would add only two methods and let Rails do the rest.

```suggestion
    def disabled
      disabled_at.present?
    end

    def disabled=(value)
      if value
        disabled_at = Time.zone.now
      else
        disabled_at = nil
      end
    end

```

Reply:
Right, I was not happy with the naming neither and waiting for some feedback, thanks.